catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AccessControlAccessPlugin.java

```diff
@@ -86,6 +95,36 @@ public UpdateRequest processPreUpdate(
           return null;
         };
 
+    /**
+     * Since this is an update request, it implies that the metacard is changing. The only way for a
+     * metacard to change is if the requesting subject has the perms to do so. The required perms
+     * require to be one of the following: <br>
+     *
+     * <ul>
+     *   <li>Access Administrator
+     *   <li>Current owner of the metacard
+     *   <li>Be explicitly placed in the R/W group on the security attributes
+     * </ul>
+     */
+    boolean foundNonWriteableMetacard =
+        input
+            .getUpdates()
+            .stream()
+            .map(Map.Entry::getValue)
+            .filter(Objects::nonNull)
+            .filter(
+                mc ->
```

Comment:
✏️ is it worth it to extract this into its own function?

Reply:
Added